### PR TITLE
Ux ui admin area improvements

### PR DESCRIPTION
**Pull request by saracoutinho**
Started working on this a while ago, figured I should finish it. 

General tidy up of the Groups and Group areas under Admin. Creating new groups and adding users to groups now happens in a modal; errors are displayed inline.

There's a bunch of slightly dubious stuff going on to get some of this working but doing it properly would require API handling changes.

### CODE REVIEW

File: src/api/fetch.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ const defaults = {
  * @returns {Promise<{result: any, success: boolean, status: number}>}
  */
 export async function fetch() {
+  // TODO - Does this arguments parsing need to be so opaque?
```
Comment (mjs) [inline src/api/fetch.js]: I've wondered about this as well! The original author of this code was an outside contributor who isn't around any more. I'm happy for this to be made more straightforward if possible.

**Comment by saracoutinho** [comment]
@mjs I finished changing the "Devices" view in the admin area. Some of the tests are failing - I fixed some but to be honest I'm not sure how to go about the rest of them. They are failing due to the changes in the "fetch" API function - it was changed to allow suppressing the global error message banner when it should be handled at a component level.

**Comment by mjs** [comment]
Thanks, I'll see if I can figure out the tests for you

**Comment by saracoutinho** [comment]
@mjs the tests are passing now - but I'm not sure what value they add.

**Comment by mjs** [comment]
> @mjs the tests are passing now - but I'm not sure what value they add.

Probably not much. I'm tempted to delete them. 

@clare is working on functional tests which will likely often much more value.

**Pull request merged**